twins/service_test.go

```diff
@@ -23,18 +23,8 @@ const (
 	natsURL    = "nats://localhost:4222"
 )
 
-func newService(tokens map[string]string) twins.Service {
-	auth := mocks.NewAuthNServiceClient(tokens)
-	twinsRepo := mocks.NewTwinRepository()
-	statesRepo := mocks.NewStateRepository()
-	idp := mocks.NewIdentityProvider()
-	subs := map[string]string{"chanID": "chanID"}
-	broker := mocks.New(subs)
-	return twins.New(broker, auth, twinsRepo, statesRepo, idp, "chanID", nil)
-}
-
 func TestAddTwin(t *testing.T) {
-	svc := newService(map[string]string{token: email})
+	svc := mocks.New(map[string]string{token: email})
```

Comment:
I didn't dig into the code logic but `svc := mocks.New(...)` sounds like you're using the mock service in `service_test.go`, that is, you're mocking the very service you want to test. If that's the case, please fix this to use real service, and mock only dependencies.

Reply:
This is the endpoints mock. I'm not mocking the service, I'm using the real service (look at the last line of the code):

```
func New(tokens map[string]string) twins.Service {
	auth := NewAuthNServiceClient(tokens)
	twinsRepo := NewTwinRepository()
	statesRepo := NewStateRepository()
	idp := NewIdentityProvider()
	subs := map[string]string{"chanID": "chanID"}
	broker := NewBroker(subs)
	return twins.New(broker, auth, twinsRepo, statesRepo, idp, "chanID", nil)
}
```

Formerly, the name of this function was `NewService`, but @manuio said that we should simply use `New` instead.

Reply:
In that case, can you, please, rename `svc` to `pub`. @manuio We do prefer `New` over `New<Suffix>`, but only in case `New` is unambiguous. In `mocks` package with multiple mocks inside it makes perfect sense to have `NewRepository` or `NewPublisher`, and in the code we'll have a nice `pub := mocks.NewPublisher(..)`.

Reply:
@dusanb94 yes, that was exactly my argument: it will be ambiguous in the code. I'm fixing it bask to `NewService`.

Reply:
@dusanb94 I agree.  As I said last time to @darkodraskovic I didn't noticed that it was a mock and I was fine to let it as it was.